lib/config.js

```diff
@@ -20,14 +22,51 @@ function Config () {
 	for (var key in config) {
 		this[key] = config[key];
 	}
+	
+	if (mode === 'worker'){
+		if (!this.output_root_dir) {
+			this.output_root_dir = '~/Guerilla';
+		}
+		this.output_root_dir = this.expandTilde(this.output_root_dir);
+		mkdirp.sync(this.output_root_dir);
+	}
+
 }
 
+
 function getUrlObj (host, port) {
 	if (host.slice(-1) === '/') host = host.slice(0, -1);
 	var urlObj = url.parse(host + ':' + port + '/');
 	return urlObj;
 }
 
+Config.prototype.getMode = function () {
+	if (argv.master) return 'master';
+	else if (argv.worker) return 'worker';
+	else throw new Error('Invalid mode.');
+}
+
+
+Config.prototype.expandTilde = function (apath) {
+	var homedir = process.env['HOME'];
+	if (!apath) return apath;
+	if (apath == '~') return homedir;
+	if (apath.slice(0, 2) != '~/') return apath;
+	return path.join(homedir, apath.slice(2));
+}
+
+Config.prototype.getGuerillaDir = function () {
+	return this.output_root_dir;
```

Comment:
See previous comment about renaming this to `Config.prototype.getDataDir` and `return this.data_dir`.